ql/instruments/bond.hpp

```diff
@@ -62,7 +62,8 @@ namespace QuantLib {
         class Price {
           public:
             enum Type { Dirty, Clean };
-            Price() : amount_(Null<Real>()) {}
+            Price() : amount_(Null<Real>()), type_(Bond::Price::Clean) {}
+            Price(Real amount) : amount_(amount), type_(Bond::Price::Clean) {}
```

Comment:
@lballabio Codacy says the constructor should be marked explicit, but I think it's totally reasonable to pass through an `int` value here and let the implicit conversion do its job.  Please let me know what your preference is and I'll adjust accordingly.

Reply:
Frankly I'd avoid taking a number without an explicit type.  I'll do it.